inbox/models/category.py

```diff
@@ -39,21 +60,28 @@ def API_OBJECT_NAME(self):
 
     # STOPSHIP(emfree): need to index properly for API filtering performance.
     name = Column(String(MAX_INDEXABLE_LENGTH), nullable=False, default='')
-    display_name = Column(String(MAX_INDEXABLE_LENGTH,
-                                 collation='utf8mb4_bin'), nullable=False)
+    display_name = Column(CategoryNameString(), nullable=False)
```

Comment:
This is really nice, but it's confusing that the new column type pre-processes values you want to compare it against, but does not pre-process values you actually assign to it. I could still say: "display_name = super-long-string", and then display_name == super-long-string would be false.

I think it'd be really cool to encapsulate both the value coercion and comparator coercion into the same type class, so we don't need to have a separate block of code below `@validates("display_name")`.

I /think/ this might work:

```
class CategoryNameString(String):
   __init__: (transformCb) =>
      this.transformCb = transformCb

   class comparator_factory(String.Comparator):
       def __eq__(self, other):
             other = Category.sanitize_name(other)
             return self.operate(operators.eq, other)

    def bind_processor(self, dialect):
        def process(value):
            return this.transformCb(super.bind_processor(dialect)(value))
        return process

```

If we could encapsulate both, we could make this more generically useful:

```
display_name = Column(StringWithTransform(MAX_INDEXABLE_LENGTH, transform=self. sanitize_name), nullable=False)

```